src/Horde.jl

```diff
@@ -1,38 +1,40 @@
 module Horde
 import Base.get
 
-export GVF, PredictionGVF, get_parameters, get,
+export GVF, GVFHorde, get_parameters, get,
     FeatureCumulant, ConstantDiscount, NullPolicy
 
 include("GVF/GVF.jl")
 
-
-mutable struct GVFHorde{T<:AbstractGVF}
-    gvfs::Array{T, 1}
+abstract type AbstractHorde end
+
+# Previous Implementation
+# struct GVFHorde{C<:AbstractCumulant, D<:AbstractDiscount, P<:AbstractPolicy} <: AbstractHorde
+#     cumulants::Vector{C}
+#     discounts::Vector{D}
+#     policies::Vector{P}
+# end
+
+# function get(gvfh::GVFHorde, state_t, action_t, state_tp1, action_tp1, preds_tp1)
+#     C = map(c -> get(c, state_t, action_t, state_tp1, action_tp1, preds_tp1), gvfh.cumulants)
+#     Γ = map(γ -> get(γ, state_t, action_t, state_tp1, action_tp1, preds_tp1), gvfh.discounts)
+#     Π_probs = map(π -> get(π, state_t, action_t, state_tp1, action_tp1, preds_tp1), gvfh.policies)
+#     return C, Γ, Π_probs
+# end
+
+# Matt's Implementation
+struct GVFHorde{T<:AbstractGVF} <: AbstractHorde
```

Comment:
Will this allow us to use a list of different GVFs? For instance:
```julia
struct GVFType1 <: AbstractGVF ...
struct GVFType2 <: AbstractGVF ...

GVFHorde([ GVFType1(), GVFType2() ])
```
This is the same concern I had with the previous implementation I believe.

Reply:
Let me test.

Reply:
Test code:

```
abstract type AbstractStruct end

Base.get(as::AbstractStruct) = as.x

struct MyStruct <: AbstractStruct
    x::Int64
end

struct MyStruct_another <: AbstractStruct
    x::Float64
end



struct StructCollection{T<:AbstractStruct}
    structs::Vector{T}
end

Base.get(asc::StructCollection) = get.(asc.structs)


function test()

    structs = [MyStruct(1), MyStruct_another(1.0)]
    println(typeof(structs))
    struct_collection = StructCollection(structs)

    println(get(struct_collection))

end
```

output:

```
julia> test()
Array{AbstractStruct,1}
Real[1, 1.0]
```